cmd/clustersynchro-manager/app/options/options.go

```diff
@@ -160,13 +162,18 @@ func (o *Options) Config() (*config.Config, error) {
 	}
 	kubeStateMetricsServerConfig := o.KubeStateMetrics.ServerConfig(metricsConfig)
 
+	if o.ShardingName != "" {
+		o.LeaderElection.ResourceName = fmt.Sprintf("%s-%s", o.LeaderElection.ResourceName, o.ShardingName)
```

Comment:
Do we need to check whether  the resource already exists here?

Reply:
The leaderelection will take care of checking for the resource, and if it already exists then it will race for it.